api/machines.go

```diff
@@ -64,6 +83,59 @@ func (mr *machinesResource) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	sendResponse(rw, http.StatusOK, page)
 }
 
+func (mr *machinesResource) patch(rw http.ResponseWriter, req *http.Request) {
+	ops := make([]machineMetadataOp, 0)
+
+	dec := json.NewDecoder(req.Body)
+
+	if err := dec.Decode(&ops); err != nil {
+		sendError(rw, http.StatusInternalServerError, err)
```

Comment:
It's not a 500-level error if the user provided data that isn't valid JSON - it should be a StatusBadRequest (400).